### DISCUSSION TITLE
Error: The specified module could not be found.

### DISCUSSION THREAD

**Discussion by gy9527** [withcatai/node-llama-cpp#319] [Q&A] [answered]
I'm developing a project using Electron. Everything works fine in development mode, but after packaging and running it, I encounter this error. I'd like to know what precautions I might have overlooked.
```
Error: The specified module could not be found.
\\?\D:\MyProjects\local-note\out\local-note-win32-x64\resources\app.asar.unpacked\node_modules\node-llama-cpp\bins\win-x64-vulkan\llama-addon.node
```
The files all exist, and the paths are correct.

**Comment by giladgd** [comment, marked as answer]
I don't have enough information to assess what could be the issue, but note that `node-llama-cpp` must not be bundled (and instead be marked as an external module) in order for it to function properly, since the file structure of it is necessary.

I recommend you to scaffold an Electron project that uses `node-llama-cpp` using this CLI command and inspect it to find what it does that your current setup doesn't do:
```
npm create --yes node-llama-cpp@beta
```

**Reply by giladgd** [reply]
@gy9527 There's an issue with the latest Electron builder version, so I've downgraded it to the latest working version in the Electron project template.
I've released a new version of `node-llama-cpp` with the fix, so try to scaffold a new project again using the aforementioned command and see whether it works for you now.

**Reply by gy9527** [reply]
thanks
